chunk/chunk_store_test.go

```diff
@@ -146,3 +146,58 @@ func diff(want, have interface{}) string {
 	})
 	return "\n" + text
 }
+
+func TestBigBuckets(t *testing.T) {
+	scenarios := []struct {
+		from, through, dailyBucketsFrom model.Time
+		buckets                         []string
+	}{
+		// Buckets are by hour until we reach the `dailyBucketsFrom`, after which they are by day.
+		{
+			from:             model.TimeFromUnix(0),
+			through:          model.TimeFromUnix(0).Add(3*24*time.Hour) - 1,
+			dailyBucketsFrom: model.TimeFromUnix(0).Add(1 * 24 * time.Hour),
+			buckets:          []string{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "d1", "d2"},
+		},
+
+		// Only the day part of `dailyBucketsFrom` matters, not the time part.
+		{
+			from:             model.TimeFromUnix(0),
+			through:          model.TimeFromUnix(0).Add(3*24*time.Hour) - 1,
+			dailyBucketsFrom: model.TimeFromUnix(0).Add(2*24*time.Hour) - 1,
+			buckets:          []string{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "d1", "d2"},
+		},
+
+		{
```

Comment:
Hmm, I guess I just wanted to show that 24 1-hour buckets disappear when you make the `dailyBucketsFriom` one day earlier than in the case above (but the one overlapping 1h bucket still remains). So maybe the following?

"// Moving `dailyBucketsFrom` to the previous day compared to the above makes 24 1-hour buckets disappear."

But yeah, the incremental value of this test case is perhaps debatable :)